File: platform/frontend/src/api/productionsandbox.ts
import { getAuthHeaders } from './auth'

const API = import.meta.env.VITE_API_URL || ''

export interface ProductionSandbox {
  id: string
  userId: string
  sandboxName: string
  provider: string
  status: string
  envVarsJson: string
  envVarCount: number
  servicesJson: string
  serviceCount: number
  region: string
  oauthConnected: boolean
  uptimeMinutes: number
  costUsd: number
  devRequestId: number | null
  createdAt: string
  updatedAt: string
}

export interface ProviderInfo {
  id: string
  name: string
  description: string
  color: string
}

export interface SandboxStats {
  totalSandboxes: number
  activeSandboxes: number
  totalEnvVars: number
  totalServices: number
  totalUptimeMinutes: number
  totalCostUsd: number
  byProvider: { provider: string; count: number; envVars: number; services: number }[]
  byStatus: { status: string; count: number }[]
}

export async function listSandboxes(provider?: string): Promise<ProductionSandbox[]> {
  const params = new URLSearchParams()
  if (provider) params.set('provider', provider)
  const res = await fetch(`${API}/api/production-sandboxes?${params}`, { headers: getAuthHeaders() })
  if (!res.ok) return []
  return res.json()
}

export async function createSandbox(req: {
  sandboxName?: string
  provider?: string
  region?: string
  devRequestId?: number
}): Promise<ProductionSandbox> {
  const res = await fetch(`${API}/api/production-sandboxes`, {
    method: 'POST',
    headers: { ...getAuthHeaders(), 'Content-Type': 'application/json' },
    body: JSON.stringify(req),
  })
  return res.json()
}

export async function stopSandbox(id: string): Promise<ProductionSandbox> {
  const res = await fetch(`${API}/api/production-sandboxes/${id}/stop`, {
    method: 'POST',
    headers: getAuthHeaders(),
  })
  return res.json()
}

export async function deleteSandbox(id: string): Promise<void> {
  await fetch(`${API}/api/production-sandboxes/${id}`, {
    method: 'DELETE',
    headers: getAuthHeaders(),
  })
}

export async function getSandboxStats(): Promise<SandboxStats> {
  const res = await fetch(`${API}/api/production-sandboxes/stats`, { headers: getAuthHeaders() })
  return res.json()
}

export async function getProviders(): Promise<ProviderInfo[]> {
  const res = await fetch(`${API}/api/production-sandboxes/providers`)
  if (!res.ok) return []
  return res.json()
}

File: platform/backend/AiDevRequest.API/Entities/ProductionSandbox.cs
namespace AiDevRequest.API.Entities;

public class ProductionSandbox
{
    public Guid Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string SandboxName { get; set; } = string.Empty;
    public string Provider { get; set; } = string.Empty; // azure, aws, vercel
    public string Status { get; set; } = "pending"; // pending, provisioning, running, stopped, error
    public string EnvVarsJson { get; set; } = "[]"; // JSON array of imported env vars (names only, not values)
    public int EnvVarCount { get; set; }
    public string ServicesJson { get; set; } = "[]"; // JSON array of connected services
    public int ServiceCount { get; set; }
    public string Region { get; set; } = string.Empty;
    public bool OAuthConnected { get; set; }
    public double UptimeMinutes { get; set; }
    public double CostUsd { get; set; }
    public int? DevRequestId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
